presto-test-jdbc-compatibility/src/test/java/io/prestosql/TestJdbcCompatibility.java

```diff
@@ -0,0 +1,479 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logging;
+import io.prestosql.plugin.mongodb.MongoPlugin;
+import io.prestosql.server.testing.TestingPrestoServer;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.sql.Array;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Strings.repeat;
+import static com.google.common.base.Verify.verify;
+import static io.prestosql.JdbcDriverCapabilities.driverVersion;
+import static io.prestosql.JdbcDriverCapabilities.hasBrokenParametricTimestampWithTimeZoneSupport;
+import static io.prestosql.JdbcDriverCapabilities.supportsParametricTimestamp;
+import static io.prestosql.JdbcDriverCapabilities.supportsParametricTimestampWithTimeZone;
+import static io.prestosql.JdbcDriverCapabilities.supportsSessionPropertiesViaConnectionUri;
+import static io.prestosql.JdbcDriverCapabilities.testedVersion;
+import static java.lang.Integer.min;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestJdbcCompatibility
+{
+    private static final Optional<Integer> VERSION_UNDER_TEST = testedVersion();
+    private static final int TIMESTAMP_DEFAULT_PRECISION = 3;
+    private static final int TIMESTAMP_JDBC_MAX_PRECISION = 9; // nanoseconds
+    private static final int TIMESTAMP_MAX_PRECISION = 12; // picoseconds
+    private static final String TESTED_TZ = "Australia/Eucla"; // GMT+8:45
+
+    private TestingPrestoServer server;
+    private String serverUrl;
+
+    @BeforeClass
+    public void setup()
+    {
+        Logging.initialize();
+
+        server = TestingPrestoServer.builder()
+                .build();
+
+        server.installPlugin(new MongoPlugin());
+
+        serverUrl = format("jdbc:presto://%s", server.getAddress());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        server.close();
+    }
+
+    @Test
+    public void ensureProperDriverVersionLoaded()
+    {
+        if (VERSION_UNDER_TEST.isEmpty()) {
+            throw new SkipException("PRESTO_JDBC_VERSION environment variable is missing");
```

Comment:
make sure to update this if you rename the env variable

maybe 

```suggestion
            throw new SkipException("Information about JDBC version under test is missing");
```